src/main.rs

```diff
@@ -78,11 +79,13 @@ async fn start_servers(
         VERSION.unwrap_or("unknown").to_string(),
         gossip.swarm.local_peer_id().to_string(),
     ));
-    let grpc_service = service.clone();
     let grpc_shutdown_tx = shutdown_tx.clone();
+    let hub_service = HubServiceServer::from_arc(service.clone())
+        .accept_compressed(CompressionEncoding::Gzip)
+        .send_compressed(CompressionEncoding::Gzip);
```

Comment:
Is this always or only if the header is present in the request? Just want to make sure it won't break existing clients.

Reply:
I'm 95% sure that it accepts it, but it doesn't require it.